### DISCUSSION TITLE
Replicas switch to file based sync instead of async

### DISCUSSION THREAD

**Discussion by kbudde** [cloudnative-pg/cloudnative-pg#3587] [General]
Just sharing some issue we recently faced and were able to fix. 
We did not find out the root cause why the cluster was not syncing as expected but at least we've found out how to bring it into the expected state.

*Symptoms:*
- Newly added cluster nodes are restoring the backup. Replication role should be `Standby (async)` but it is `Standby (file based)`
- Checking replication slot settings: The name of the database is wrong in `primary_conninfo` and `primary_slot_name`

*Checks and fixes:*
Check and fix config files on all cluster nodes:
```bash 
cd /var/lib/postgresql/data/pgdata/

cat postgresql.auto.conf # this file should not contain anything
# echo >postgresql.auto.conf to clear the content 

cat postgresql.conf # this file should have two includes at the end: custom.conf and override.conf -> in our case it was missing.
# echo "include 'override.conf'" >>postgresql.conf # add the include to the end if it's mssing

psql -c 'select pg_reload_conf();' #  Sync should switch to async immidiately
```

important: Check the configuration on all nodes; especially on the primary. Then it will persist to new nodes. 

As said in the beginning, we have no clue 🤷🏽  how we were able to not have a good default configuration. The above settings were just changing the configuration back to CNPG defaults and then it works.

Thank you @leonardoce for your help in slack: https://cloudnativepg.slack.com/archives/C03AX0J5P29/p1702624524586329

**Comment by ecksun** [comment]
I have run into the very same issue with the one difference that `postgresql.conf` does include `include 'override.conf'`. However my understanding is that `postgresql.auto.conf` is the last file to be loaded anyway:

From [the documentation](https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-CONFIGURATION-FILE):

>  Settings in postgresql.auto.conf override those in postgresql.conf.

I think this is a bug, probably introduced in 1.21.1 so I created an issue for it: https://github.com/cloudnative-pg/cloudnative-pg/issues/3588

I have intentionally not tried to fix my cluster in the hopes that it can help find and fix the issue in CNPG
